ptxcompiler/patch.py

```diff
@@ -122,6 +129,11 @@ def get_cubin(self, cc=None):
 
 
 def patch_needed():
+    # If Numba is not present, we don't need the patch.
+    # We also can't use it to check driver and runtime versions, so exit early.
+    if not _numba_version_ok:
+        return False
+
```

Comment:
I thought about generating a warning but then I was concerned that the warning would potentially show up "out of nowhere" for users of packages that depend on ptxcompiler, with little they could do about it.

I didn't use the logger from `get_logger()`, because ptxcompiler's patching tries to use Numba's logging settings, and that will fail if Numba is not present.

Reply:
Ah makes sense. thanks.